Userland/Libraries/LibWeb/CSS/StyleProperties.cpp

```diff
@@ -139,6 +139,13 @@ CSS::Size StyleProperties::size_value(CSS::PropertyID id) const
         switch (value->to_keyword()) {
         case Keyword::Auto:
             return CSS::Size::make_auto();
+        case Keyword::Normal:
+            if (id == CSS::PropertyID::Gap || id == CSS::PropertyID::ColumnGap || id == CSS::PropertyID::RowGap) {
+                // FIXME: Context based value, assign `1em` if container has multiple columns:
+                // https://drafts.csswg.org/css-align/#valdef-row-gap-normal
+                return CSS::Size::make_px(0);
+            }
+            VERIFY_NOT_REACHED();
```

Comment:
This doesn't seem correct. The spec says this should affect the *used* value, but by changing it here, we are altering the *computed* value.

Another issue is that we're using `CSS::Size` to represent gap values in the engine, despite not actually needing the full range of possible `CSS::Size` values. I believe we're using the `auto` state of `CSS::Size` to represent `normal`, since it's the only possible non-length value.

Reply:
Agreed, a `Size` doesn't make sense here. IMO the `gap` properties should be stored in `StyleProperties` as `Optional<LengthPercentage>` with the empty state meaning `normal`.